### PR TITLE
I1 ship

### PR DESCRIPTION
**Pull request by AuroraTuring**
Add attr_reader, Ship class, initialize data. Add #sunk? and #hit methods.

### CODE REVIEW

File: lib/ship.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Ship
+  attr_reader :name, :length, :health
+
+  def initialize(name, length)
+    @name = name
+    @length = length
+    @health = 3
+    @sunk = false
```
Comment (NeilTheSeal) [inline lib/ship.rb]: We don't need the @sunk attribute - see comment on line 11

File: lib/ship.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def sunk?
```
Comment (NeilTheSeal) [inline lib/ship.rb:10]: This method can be shortened to a single line -
`@health <= 0`

File: lib/ship.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    if @health == 0
+      @sunk = true
+    else
+      @sunk
+    end
+  end
+
+  def hit
```
Comment (NeilTheSeal) [inline lib/ship.rb:19]: We need an `if` statement to only subtract health `if health > 0` so we don't ever get a negative value for health

**Review by NeilTheSeal — CHANGES_REQUESTED** [review]

File: lib/ship.rb (hunk 2 above)
Comment (AuroraTuring) [reply lib/ship.rb:10]: I have updated the code!

**Pull request merged**